py3status/modules/nvidia_temp.py

```diff
@@ -32,9 +38,21 @@ class Py3status:
     """
     # configuration parameters
     cache_timeout = 10
-    format_prefix = "GPU: "
-    format_units = u"°C"
-    temp_separator = '|'
+    format = "GPU: {format_temp}"
+    format_temp = u"{temp}°C"
+    icon_separator = '|'
+    msg_unavailable = "GPU: N/A"
+
+    class Meta:
+        deprecated = {
+            'rename': [
+                {
+                    'param': 'temp_separator',
+                    'new': 'icon_separator',
+                    'msg': 'obsolete parameter use `icon_separator`',
+                },
+            ],
+        }
```

Comment:
`format_units` needs to be dealt with.  It's a bit of a pain but basically we will use 'function' to create a new `format_temp` (unless the user supplied one) and then remove `format_units`